### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by jim-minter**
remove CheckExistence API call and fix possible logic misordering in delete
standardise autorest.DetailedError error handling

### CODE REVIEW

File: pkg/backend/openshiftcluster/delete.go [hunk 1]
```diff
@@ -54,22 +54,17 @@ func (m *Manager) Delete(ctx context.Context) error {
...
 	if err != nil {
 		return err
 	}
 
 	m.log.Printf("deleting resource group %s", m.doc.OpenShiftCluster.Properties.ResourceGroup)
-	return m.groups.DeleteAndWait(ctx, m.doc.OpenShiftCluster.Properties.ResourceGroup)
+	err = m.groups.DeleteAndWait(ctx, m.doc.OpenShiftCluster.Properties.ResourceGroup)
+	if detailedErr, ok := err.(autorest.DetailedError); ok &&
+		detailedErr.StatusCode == http.StatusForbidden {
+		err = nil
```
Comment (m1kola) [inline pkg/backend/openshiftcluster/delete.go]: Could you please explain why we say that there is no error if we can't remove the resource group?

Comment (jim-minter) [reply pkg/backend/openshiftcluster/delete.go]: Yes: backend functions should be restartable.  Say a delete is running, but that RP crashes and restarts: by the time it gets here, the group could be gone.  If so, that shouldn't be an error.

I'm not currently 100% sure if the error returned in dev == the error in prod, and I'm also not sure if to expect forbidden or notfound - will check as a followup

Comment (m1kola) [reply pkg/backend/openshiftcluster/delete.go]: Got it. Thanks.

Comment (jim-minter) [reply pkg/backend/openshiftcluster/delete.go]: Yes, slightly non-intuitively, it's Forbidden.

**Pull request merged**